pkg/asset/agent/image/ignition.go

```diff
@@ -31,16 +74,107 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 	agentManifests := &manifests.AgentManifests{}
 	dependencies.Get(agentManifests)
 
-	configBuilder, err := imagebuilder.New(*agentManifests)
+	pullSecret := agentManifests.PullSecret
+	infraEnv := agentManifests.InfraEnv
+
+	config := igntypes.Config{
+		Ignition: igntypes.Ignition{
+			Version: igntypes.MaxVersion.String(),
+		},
+		Passwd: igntypes.Passwd{
+			Users: []igntypes.PasswdUser{
+				{
+					Name: "core",
+					SSHAuthorizedKeys: []igntypes.SSHAuthorizedKey{
+						igntypes.SSHAuthorizedKey(infraEnv.Spec.SSHAuthorizedKey),
+					},
+				},
+			},
+		},
+	}
+
+	if pullSecret.StringData[".dockerconfigjson"] == "" {
+		return fmt.Errorf("pull secret is missing")
+	}
+
+	nodeZeroIP, err := manifests.GetNodeZeroIP(agentManifests.NMStateConfigs)
+	if err != nil {
+		return err
+	}
+
+	agentTemplateData := getTemplateData(
+		pullSecret.StringData[".dockerconfigjson"],
```

Comment:
Could be retrieved from the method added in [PR 5943](https://github.com/openshift/installer/pull/5943/files#diff-ed3ddff72757f2b9f8078d820e70ac801ad04c31885e033d6722ac6ae865f10eR92)